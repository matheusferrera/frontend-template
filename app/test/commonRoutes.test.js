import React from "react";
import renderer from "react-test-renderer";

import { describe, expect, it, jest } from "@jest/globals";

import FAQ from "../src/pages/FAQ";
import NotFound from "../src/pages/NotFound";
import { TestWrapper } from "./testWrapper";

describe("Teste de componentes", () => {
  it("Not found renderizada corretamente", () => {
    const component = renderer.create(<NotFound />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("PÃ¡gina de FAQ", () => {
    const component = renderer.create(
      <TestWrapper>
        <FAQ />
      </TestWrapper>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    jest.resetAllMocks();
  });
});
