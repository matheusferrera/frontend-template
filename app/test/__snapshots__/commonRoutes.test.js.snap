// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de componentes Not found renderizada corretamente 1`] = `
<div>
  <h1>
    404 - Página não encontrada
  </h1>
  <p>
    Ops! A página que você está procurando não existe.
  </p>
</div>
`;

exports[`Teste de componentes Página de FAQ 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1l0w7oi-MuiContainer-root"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 css-ywhdiz-MuiTypography-root"
  >
    Perguntas Frequentes - FAQ
  </h4>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      O que é o Programa Redução da Pobreza e o que ele oferece?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      O Programa Redução da Pobreza é um conjunto de ações destinadas a auxiliar as pessoas inscritas no Cadastro Único, preferencialmente aquelas com renda familiar de até ½ salário-mínimo por pessoa. Por meio do site do Programa Redução da Pobreza, você encontra cursos gratuitos de qualificação profissional nas modalidades presencial e de Ensino à Distância (EaD), possibilidade de acesso ao microcrédito produtivo orientado para investir no seu próprio negócio e também pode no próprio site, elaborar um currículo e se candidatar a vagas de emprego.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Se você já estiver no Cadastro Único e quiser participar do Programa Redução da Pobreza, acesse o site clicando
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza  "
        onBlur={[Function]}
        onFocus={[Function]}
      >
        aqui
      </a>
      e faça seu cadastro.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Como faço meu cadastro no Portal Programa Redução da Pobreza?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Para fazer o cadastro no Programa Redução da Pobreza, é necessário estar inscrito no Cadastro Único. Se você está inscrito no Cadastro Único, o seu CPF será identificado pelo sistema no momento que você estiver fazendo o seu cadastro no site Programa Redução da Pobreza. Para isso, acesse o site
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        aqui
      </a>
      e clique nos botões Quero aprender, Quero empreender ou Quero emprego”. Ao acessar essa aba você é direcionado para a Conta de acesso única do Governo &lt;
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-k985qi-MuiTypography-root-MuiLink-root"
        href="http://www.acesso.gov.br/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        acesso.gov.br/
      </a>
      &gt; para autenticar telefone e/ou e-mail preenchendo os dados solicitados. Depois desse cadastro o sistema irá informar se você está inscrito no Cadastro Único. Se você não estiver no Cadastro Único, procure o setor responsável pelo Cadastro Único em seu município, ou um Centro de Referência de Assistência Social (CRAS). Para fazer parte do Cadastro Único, é necessário ter renda mensal igual ou inferior a meio salário-mínimo por pessoa ou renda familiar mensal de até três salários mínimos.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      O Programa Redução da Pobreza oferece oportunidades de qualificação, de emprego e de renda e promove a autonomia das pessoas inscritas no Cadastro Único.
    </p>
    <ol>
      <li>
        Acessar o site clicando
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
          href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          aqui
        </a>
      </li>
      <li>
        Clicar em uma das opções
      </li>
      <div
        className="MuiBox-root css-r9ni7l"
      >
        <img
          src="test-file-stub"
          style={
            {
              "width": 700,
            }
          }
        />
      </div>
      <li>
        Informar o seu CPF, selecionar estado e acionar a opção “Não sou robô”
      </li>
      <div
        className="MuiBox-root css-159z2lj"
      >
        <img
          src="test-file-stub"
          style={
            {
              "width": 400,
            }
          }
        />
      </div>
      <li>
        Acionar opção (GOV.BR)
      </li>
      <div
        className="MuiBox-root css-159z2lj"
      >
        <img
          src="test-file-stub"
        />
      </div>
      <li>
        Informar CPF e senha no GOV.BR e caso usuário não tenha cadastro no GOV.BR deverá realizar um cadastro; 
      </li>
      <div
        className="MuiBox-root css-4kdwyk"
      >
        <img
          src="test-file-stub"
          style={
            {
              "width": 600,
            }
          }
        />
      </div>
      <li>
        Validar e-mail e/ou número de telefone;
      </li>
      <div
        className="MuiBox-root css-r9ni7l"
      >
        <img
          src="test-file-stub"
          style={
            {
              "width": 700,
            }
          }
        />
      </div>
      <li>
        Aceitar termo de acesso ao PROGRAMA REDUÇÃO DA POBREZA.
      </li>
    </ol>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      OBS: Em caso de o usuário ter esquecido o e-mail cadastrado ele deve seguir as seguintes instruções, retirado do site dúvidas frequentes do GOV.BR clicando no
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fxugtv-MuiTypography-root-MuiLink-root"
        href="https://faq-login-unico.servicos.gov.br/en/latest/_perguntasdafaq/naotenhoacessoaoemailcadastradonologin.html"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        link
      </a>
      .
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Para alterar o e-mail cadastrado, a Administração Pública precisa ter certeza de que você é você mesmo. Há diferentes modos de fazer isso:
    </p>
    <ul>
      <li>
        Validação facial no aplicativo gov.br (para quem possui carteira de motorista – CNH ou Título de Eleitor com biometria cadastrada - Tribunal Superior Eleitoral)
      </li>
      <li>
        Utilização de internet banking (para quem possui conta em um dos seguintes bancos: Banco do Brasil, Banrisul, Banese, Bradesco, Caixa Econômica Federal, Itaú, Santander ou Sicoob)
      </li>
      <li>
        Utilização do autoatendimento dos Bancos Credenciados
      </li>
    </ul>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Caso não se encaixe em nenhuma dessas situações, preencha o formulário Como Podemos te ajudar? com as informações:
    </p>
    <ol>
      <li>
        Preencher no campo assunto: Alteração do e-mail por solicitação do cidadão
      </li>
      <li>
        Preencher o conteúdo com as informações:
        <ul>
          <li>
            Serviço que deseja acessar com Login Único
          </li>
          <li>
            E-mail para alteração;
          </li>
          <li>
            Anexar arquivo com o Documento de Identificação e autofotografia de seu rosto, segurando o documento original com foto mais recente possível para checagem;
          </li>
          <li>
            Seguinte texto: Eu, nome do cidadão, portador do CPF, CPF do cidadão , autorizo o Ministério da Gestão e da Inovação em Serviços Públicos a alterar meus dados cadastrais da plataforma de autenticação Login Único (gov.br). Cidade residência do cidadão , data do preenchimento da autorização.
          </li>
        </ul>
      </li>
    </ol>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Logo que o Ministério da Gestão e da Inovação em Serviços Públicos receber a solicitação, será realizada a análise de segurança e confirmação será encaminhada para email solicitado para alteração. Ausência das informações acima impossibilita análise e retorno.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Eu quero trabalhar/ Quero um emprego. Como o Programa Redução da Pobreza pode me ajudar?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Primeiramente, você deve se cadastrar no Portal Programa Redução da Pobreza. Para isso, acesse o site clicando
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza  "
        onBlur={[Function]}
        onFocus={[Function]}
      >
        aqui
      </a>
      e clique em um dos botões 
      <b>
        QUERO APRENDER, QUERO EMPREENDER, QUERO EMPREGO
      </b>
      . Ao acessar essa aba você será direcionado para a Conta de acesso única do Governo &lt;
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-k985qi-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        acesso.gov.br/
      </a>
      &gt; para autenticar telefone e/ou e-mail preenchendo os dados solicitados. IMPORTANTE! Para se cadastrar no Programa Redução da Pobreza é necessário estar inscrito no Cadastro Único. Ao preencher as informações solicitadas, será possível se candidatar a vagas de emprego, de estágio ou de jovem aprendiz em sua região, de acordo com o seu perfil profissional. Após o cadastramento inicial, clique no botão 
      <b>
        “Vagas de Trabalho”
      </b>
      , depois em 
      <b>
        “Meu Currículo”
      </b>
      .
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      No Portal Programa Redução da Pobreza você poderá:
    </p>
    <ul>
      <li>
        Criar um currículo para você;
      </li>
      <li>
        Enviar seu currículo diretamente a empresas ofertantes de vagas de trabalho (se for uma vaga cadastrada diretamente pela empresa) ou acessar plataformas online parceiras do Programa Redução da Pobreza para se candidatar a uma vaga;
      </li>
      <li>
        Verificar se há oportunidades de trabalho em sua região;
      </li>
      <li>
        Verificar se alguma oportunidade de trabalho corresponde ao seu perfil profissional;
      </li>
      <li>
        Gravar seu currículo em formato PDF;
      </li>
      <li>
        Realizar a impressão do currículo;
      </li>
      <li>
        Se inscrever e participar de cursos presenciais e à distância (EaD), gratuitos;
      </li>
    </ul>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Há cada vez mais empresas divulgando oportunidades de trabalho no Portal Programa Redução da Pobreza. A utilização dessa ferramenta por empresas é resultado da articulação dos poderes públicos do seu município, juntamente com o Ministério do Desenvolvimento e Assistência Social, Família e combate à Fome.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Já estou empregado(a). Posso me cadastrar no Programa Redução da Pobreza?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Sim! Mesmo se você estiver empregado(a), é possível se cadastrar no portal do Programa Redução da Pobreza para ter acesso a vagas de emprego, estágio ou Jovem Aprendiz, cursos de qualificação profissional e empreendedorismo na sua região.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Como posso atualizar as informações do meu currículo?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Para atualizar seu currículo, entre no site clicando
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fxugtv-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza  "
        onBlur={[Function]}
        onFocus={[Function]}
      >
        aqui
      </a>
      , clique nos botões QUERO APRENDER, QUERO EMPREENDER, QUERO EMPREGO. Depois é só inserir as informações solicitadas para acessar o Portal. Em seguida, na coluna à esquerda, clique em “
      <b>
        Vagas de Trabalho
      </b>
      ”, depois em “
      <b>
        Meu Currículo
      </b>
      ” e em seguida, na parte superior da tela, clique em “
      <b>
        Alterar Informações
      </b>
      ”. Depois que fizer as alterações ou atualizações, clique em “
      <b>
        salvar
      </b>
      ”.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Como é feita a seleção de vagas?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      As empresas e instituições parceiras do Programa Redução da Pobreza podem ofertar vagas de trabalho (na forma de contrato de emprego, estágio ou jovem aprendiz) e vagas de cursos gratuitos de qualificação. O candidato a uma vaga de emprego, estágio ou Jovem Aprendiz, ao acessar o Portal Programa Redução da Pobreza clicando
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fxugtv-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza  "
        onBlur={[Function]}
        onFocus={[Function]}
      >
        aqui
      </a>
      , pode visualizar oportunidades de trabalho e cursos disponíveis em sua região; e, se já tiver um currículo cadastrado, pode manifestar interesse por uma vaga anunciada, ou pode ser contatado diretamente pela empresa para uma entrevista, mesmo sem ter manifestado o interesse na vaga.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Vale lembrar que nenhuma vaga de emprego, estágio ou Jovem Aprendiz é exclusiva para o público do Programa Redução da Pobreza. Assim, o candidato participará de processo seletivo e a empresa não é obrigada a contratar o público do Programa Redução da Pobreza.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Muitos trabalhadores já foram contratados por empresas que buscaram candidatos no Portal Programa Redução da Pobreza.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      O Ministério do Desenvolvimento e Assistência Social, Família e combate à Fome não se responsabiliza pela contratação em qualquer que seja a modalidade, em nenhuma fase. O Programa Redução da Pobreza opera como um mediador entre as empresas ofertantes de vaga de trabalho e trabalhadores à procura de oportunidades de trabalho.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Quais são as ações de Qualificação no Programa Redução da Pobreza?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      A Qualificação compreende cursos de:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      a). Educação Financeira; esta ação está disponível no Portal Capacitação MDS:
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1fxugtv-MuiTypography-root-MuiLink-root"
        href="https://novoead.cidadania.gov.br/index  "
        onBlur={[Function]}
        onFocus={[Function]}
      >
        novoead.cidadania.gov.br/index
      </a>
      ;
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      b). Inclusão Digital;
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      c). Cursos disponibilizados pelos parceiros em diversas áreas de conhecimento;
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      d). Cursos do sistema S.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Do que se trata a área de Educação Financeira?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      O interessado nesta formação poderá acessar dois ambientes para a oferta deste curso:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      <b>
        Cursos de Gestão de Finanças Pessoais e Educação Financeira nas Escolas
      </b>
      , que já está disponibilizado no Portal Programa Redução da Pobreza e o interessado poderá acessar seguindo os passos abaixo:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Entre no site
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-cuw6m2-MuiTypography-root-MuiLink-root"
        href="https://www.gov.br/mds/pt-br/acoes-e-programas/programa-reducao-da-pobreza  "
        onBlur={[Function]}
        onFocus={[Function]}
      >
        aqui
      </a>
      ;
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Clique no botão QUERO APRENDER;
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Na plataforma do Programa Redução da Pobreza clique em “Cursos”. Você verá na tela uma lista de oferta de cursos gratuitos em várias áreas de conhecimento. O curso identificado está nas trilhas de cursos ofertados, onde há os cursos de Gestão de Finanças Pessoais e Educação Financeira nas Escolas.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      <b>
        Curso EaD Educação Financeira para Empreendedores Populares
      </b>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Ofertado pelo portal capacitação MDS:
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1qx9wka-MuiTypography-root-MuiLink-root"
        href="https://novoead.cidadania.gov.br/admin/selection/inforequest/312668"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        novoead.cidadania.gov.br/admin/selection/inforequest/312668
      </a>
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Gostaria de fazer um curso no Programa Redução da Pobreza, mas não consigo acessar a plataforma porque não tenho computador. O que faço?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      O Programa Redução da Pobreza é uma plataforma digital e que, necessariamente, precisa ser acessada por meio eletrônico. Orientamos que recorra ao CRAS mais próximo de seu domicílio para verificar possíveis cursos ofertados on-line ou presenciais em sua região. Um dispositivo com internet é necessário não só para realizar as inscrições, mas também para fazer o curso propriamente dito.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      O cidadão deve, necessariamente, estar inscrito no Cadastro Único Para Programas Sociais do Governo Federal, requisito obrigatório para acessar o Programa Redução da Pobreza.
    </p>
  </div>
  <div
    className="MuiBox-root css-v57q9y"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1ntuugf-MuiTypography-root"
    >
      Se eu participar do Programa Redução da Pobreza e minha renda aumentar, eu perco o benefício do Bolsa Família?
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Não, você não deixará de receber o Bolsa Família ao participar do Programa Redução da Pobreza
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1no9ono-MuiTypography-root"
    >
      Importante esclarecer que o Programa Bolsa Família possui regras de funcionamento que não estão vinculadas com o Programa Redução da Pobreza. O Programa Bolsa Família verifica a renda familiar por meio de entrevista realizada no CRAS quando da atualização do Cadastro Único da família. Portanto, um aumento de renda, sobretudo aquela advinda de atividades informais ou que variam de um mês para o outro, não implicará a perda do benefício do Programa Bolsa Família.
    </p>
  </div>
</div>
`;
